common/firmware/head-ff/spi/spi.c

```diff
@@ -0,0 +1,243 @@
+        */
+        HAL_GPIO_DeInit(GPIOB,
+                        GPIO_PIN_12 | GPIO_PIN_13 | GPIO_PIN_14 | GPIO_PIN_15);
+        HAL_GPIO_DeInit(GPIOC, GPIO_PIN_6 | GPIO_PIN_7);
+    }
+    else if (hspi->Instance == SPI3) {
+        /* Peripheral clock disable */
+        __HAL_RCC_SPI2_CLK_DISABLE();
+
+        /**SPI3 GPIO Configuration
+        PA04     ------> SPI3_NSS
+        PC10     ------> SPI3_SCK
+        PC11     ------> SPI3_MISO
+        PC12     ------> SPI3_MOSI
+
+        Step/Dir
+         PC1  ---> Dir Pin Motor on SPI3 (Z-axis)
+         PC0  ---> Step Pin Motor on SPI3 (Z-axis)
+        */
+        HAL_GPIO_DeInit(GPIOC,
+                        GPIO_PIN_10 | GPIO_PIN_11 | GPIO_PIN_12 | GPIO_PIN_0 | GPIO_PIN_1);
+        HAL_GPIO_DeInit(GPIOA, GPIO_PIN_4);
+    }
+}
+
+
+/**
+ * @brief SPI2 Initialization Function
+ * @param None
+ * @retval None
+ */
+SPI_HandleTypeDef MX_SPI2_Init() {
+    /* SPI2 parameter configuration*/
+    __HAL_RCC_SPI2_CLK_ENABLE();
+    SPI_HandleTypeDef hspi2 = {
+        .Instance = SPI2,
+        .Init = {.Mode = SPI_MODE_MASTER,
+                 .Direction = SPI_DIRECTION_2LINES,
+                 .DataSize = SPI_DATASIZE_8BIT,
+                 .CLKPolarity = SPI_POLARITY_HIGH,
+                 .CLKPhase = SPI_PHASE_2EDGE,
+                 .NSS = SPI_NSS_SOFT,
+                 .BaudRatePrescaler = SPI_BAUDRATEPRESCALER_32,
+                 .FirstBit = SPI_FIRSTBIT_MSB,
+                 .TIMode = SPI_TIMODE_DISABLE,
+                 .CRCCalculation = SPI_CRCCALCULATION_DISABLE,
+                 .CRCPolynomial = 7,
+                 .CRCLength = SPI_CRC_LENGTH_DATASIZE,
+                 .NSSPMode = SPI_NSS_PULSE_DISABLE}
+
+    };
+
+    if (HAL_SPI_Init(&hspi2) != HAL_OK) {
+        Error_Handler();
+    }
+    return hspi2;
+}
+
+/**
+ * @brief SPI3 Initialization Function
+ * @param None
+ * @retval None
+ */
+SPI_HandleTypeDef MX_SPI3_Init() {
+    /* SPI2 parameter configuration*/
+    __HAL_RCC_SPI3_CLK_ENABLE();
+    SPI_HandleTypeDef hspi3 = {
+        .Instance = SPI3,
+        .Init = {.Mode = SPI_MODE_MASTER,
+                 .Direction = SPI_DIRECTION_2LINES,
+                 .DataSize = SPI_DATASIZE_8BIT,
+                 .CLKPolarity = SPI_POLARITY_HIGH,
+                 .CLKPhase = SPI_PHASE_2EDGE,
+                 .NSS = SPI_NSS_SOFT,
+                 .BaudRatePrescaler = SPI_BAUDRATEPRESCALER_32,
+                 .FirstBit = SPI_FIRSTBIT_MSB,
+                 .TIMode = SPI_TIMODE_DISABLE,
+                 .CRCCalculation = SPI_CRCCALCULATION_DISABLE,
+                 .CRCPolynomial = 7,
+                 .CRCLength = SPI_CRC_LENGTH_DATASIZE,
+                 .NSSPMode = SPI_NSS_PULSE_DISABLE}
+
+    };
+
+    if (HAL_SPI_Init(&hspi3) != HAL_OK) {
+        Error_Handler();
+    }
+    return hspi3;
+}
+
+void SPI2_init() {
+    handle_SPI2 = MX_SPI2_Init();
+}
+
+void SPI3_init() {
+    handle_SPI3 = MX_SPI3_Init();
+}
+
+void Set_CS_Pin() { 
```

Comment:
why do we set both here? i think we probably want these interfaces fully separate